product_internal_reference_generator/models/product_template.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Ooops - Ilyas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    int_ref_template_id = fields.Many2one(
+        "product.code.sequence", "Internal Reference Template"
+    )
+    variants_sequence_id = fields.Many2one("ir.sequence")
+    variants_prefix = fields.Char(
+        "Internal Reference Prefix", readonly=True, tracking=True
+    )
+    default_code_editable = fields.Boolean(compute="_compute_default_code_editable")
+
+    def _compute_default_code_editable(self):
+        for rec in self:
+            if self.env.user.has_group(
+                "product_internal_reference_generator.group_int_ref_template_always_visible"
+            ):
+                rec.default_code_editable = True
+            else:
+                rec.default_code_editable = not bool(rec.default_code)
```

Comment:
**suggestion**: decorate with `@api.depends_context('uid')`

**suggestion(non-blocking)**: couldn't this have also been achieved by having the field twice with
`groups="product_internal_reference_generator.group_int_ref_template_always_visible"` and
`groups="!product_internal_reference_generator.group_int_ref_template_always_visible"` ?